main/src/ca/uwaterloo/flix/language/errors/ResolutionError.scala

```diff
@@ -1207,6 +1207,23 @@ object ResolutionError {
     }
   }
 
+  /**
+   * An error raised to indicate a `put` struct expression attempts to modify an immutable field
+   *
+   * @param loc    the location where the error occurred
+   */
+  case class IllegalPut(loc: SourceLocation) extends ResolutionError with Recoverable {
+    override def summary: String = s"Modification of struct field not marked as `mut`"
```

Comment:
How about:

Modification of immutable field. Mark the field as `mut` to allow mutation.